src/shared/cpu-set-util.c

```diff
@@ -133,6 +133,28 @@ int cpu_set_add_all(CPUSet *a, const CPUSet *b) {
         return 1;
 }
 
+int cpu_set_copy(CPUSet *dst, CPUSet *src) {
+        cpu_set_t *t;
+
+        assert(src);
+        assert(dst);
+
+        if(src->set == NULL || src->allocated == 0)
+                return -EFAULT;
+
+        /* ensure src and dst are the same size */
+        t = realloc(dst->set, src->allocated);
```

Comment:
hmm, in most cases we'd assume a copy call would never read from the target, but only write to it. with this realloc() you however assume that the target has been initialized somehow before. please don't do that. please fully reset the target, do not read from it.